packages/adapter/src/methods.ts

```diff
@@ -0,0 +1,154 @@
+import {SnapConfig} from "@nodefactory/metamask-polkadot-types";
+import {BlockInfo} from "../../../packages/snap/src/rpc/substrate/getBlock";
+
+export async function getAccountAddress(pluginOrigin: string): Promise<string> {
+  return await window.ethereum.send({
+    method: pluginOrigin,
+    params: [{
+      method: 'getAddress'
+    }]
+  }) as string;
+}
+
+export function hasMetaMask(): boolean {
+  if (!window.ethereum) {
+      return false
+  }
+  return window.ethereum.isMetaMask;
+
+}
+
+export async function installPolkadotSnap(pluginOrigin: string): Promise<boolean> {
+  try {
+    await window.ethereum.send({
+      method: 'wallet_enable',
+      params: [{
+        [pluginOrigin]: {}
+      }]
+    });
+    return true;
+  } catch (e) {
+    console.log("Failed to install snap", e);
+    return false;
+  }
+}
+
+export async function isPolkadotSnapInstalled(pluginOrigin: string): Promise<boolean> {
+  try {
+    const result = await window.ethereum.send({
+      method: 'wallet_getPlugins',
+    }) as {[k: string]: object};
+    return !!result[pluginOrigin];
+  } catch (e) {
+    console.log("Failed to obtain installed plugins", e);
+    return false;
+  }
+}
+
+export async function addKusamaAsset(pluginOrigin: string): Promise<void> {
+  await window.ethereum.send({
+      method: pluginOrigin,
+      params: [{
+          method: 'addKusamaAsset'
+      }]
+  });
+}
+
+export async function addPolkadotAsset(pluginOrigin: string): Promise<void> {
+  await window.ethereum.send({
+      method: pluginOrigin,
+      params: [{
+          method: 'addPolkadotAsset'
+      }]
+  });
+}
+
+async function sendSnapMethod(method: 'getBalance' | 'getAddress' | 'getPublicKey' | 'exportSeed', pluginOrigin: string): Promise<string|null> {
+  try {
+      await window.ethereum.send({
+          method: pluginOrigin,
+          params: [{
+              method: method
+          }]
+      });
+  } catch (e) {
+      console.log("Keys not generated", e);
+  }
+  return null;
+}
+
+export async function getBalance(pluginOrigin: string): Promise<string> {
+  const response = await sendSnapMethod("getBalance", pluginOrigin);
+  return (response != null) ? response : "Unable to fetch balance";
+}
+
+export async function getAddress(pluginOrigin: string): Promise<string> {
+  const response = await sendSnapMethod("getAddress", pluginOrigin);
+  return (response != null) ? response : "Unable to fetch address";
```

Comment:
Didn't mean you to return string to user, just let error bubble up and user will catch it and handle it